### ISSUE TITLE
.psc-ide-port sometimes goes "stale", causes silent IDE failures until deletion

### ISSUE THREAD

**Issue by cemerick** [nwolverson/atom-ide-purescript#196]
At some point yesterday, my purescript-related IDE capabilities disappeared: no tooltips, no go-to-definition, no completions, issuing build commands had no effect in buffers or on disk.

I flailed around for some time trying to diagnose the problem, eventually noticing the `.psc-ide-port` file. I then noticed that the port number therein did not change across an Atom restart. I deleted the file, and all of the above functionality returned after issuing a "restart" action.

Since then, the port file has gone stale one additional time, again fixed by removing the port file and restarting the IDE server. No notices appeared in the log indicating that anything had failed or gone awry in any way.

**Comment by nwolverson** [comment]
If the atom plugin didn't create the port file (eg it was created by another editor or pscid) it won't delete it on close as it usually should; I think if at startup it can't actually connect you should see an error or it stating it's not being used (check dev tools console I think)

First thing to check if stuff stops working, is `purs ide` actually running at the time?

**Comment by cemerick** [comment]
I'm not using any other editor for Purescript, and I'm not using pscid.

I kept a pretty close eye on the dev tools console, which I've made as verbose as possible. When this problem occurs, there are no messages there, and none via the atom UI chrome.

I'm starting to wonder if it has something to do with having multiple project folders; I removed one, which seems to correlate with when the issue occurred last.

Closing for now.
